fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -114,43 +114,18 @@ fn expand_function_arguments(
     custom_enums: &HashMap<String, Property>,
     custom_structs: &HashMap<String, Property>,
 ) -> Result<(TokenStream, TokenStream), Error> {
-    let n_inputs = fun.inputs.len();
-    // Check that we have the expected types
-    if n_inputs != 4 {
-        return Err(Error::MissingData(format!(
-            "Expected exactly 4 inputs, found {}",
-            n_inputs
-        )));
-    }
-    let given_types: Vec<String> = fun.inputs[..3]
-        .iter()
-        .map(|x| x.type_field.clone())
-        .collect();
-    let expected = ["u64", "u64", "b256"];
-    if given_types != expected {
-        return Err(Error::InvalidType(format!(
-            "Expected the 3 first types to be {:?}, found {:?}",
-            expected, given_types
-        )));
-    };
     // Ignore the first three arguments so we don't have to provide them when calling
     // the contracts' methods
-    let first_index = 3;
     let mut args = Vec::with_capacity(1);
```

Comment:
keep the line 117 and use `n_inputs` as vec capacity (instead of 1)

Reply:
Uh... We don't need `with_capacity`at all. Was trying to preserve as much as I could, but this isn't really necessary anymore.

Reply:
ok!